src/Dafda/Serializing/DefaultPayloadSerializer.cs

```diff
@@ -49,16 +51,20 @@ private object ConvertToMessagePayload(PayloadDescriptor descriptor)
             var envelope = new Dictionary<string, object>
             {
                 {
-                    MakeKeyFrom("MessageId"), 
+                    MakeKeyFrom("MessageId"),
                     descriptor.MessageId
                 },
                 {
-                    MakeKeyFrom("Type"), 
+                    MakeKeyFrom("Type"),
                     descriptor.MessageType
                 }
             };
 
-            foreach (var (key, value) in descriptor.MessageHeaders)
+            var messageHeaders = descriptor.MessageHeaders
+                .Where(k => !string.Equals(k.Key, "messageId", StringComparison.InvariantCultureIgnoreCase))
+                .Where(k => !string.Equals(k.Key, "type", StringComparison.InvariantCultureIgnoreCase));
```

Comment:
I like it better to keep the logic in the serializer 👍

But I don't really know why these headers are filtered out from reading (just this part of) the code. Would a named list describing what headers are already defined/default (or better naming), using this to filter out entries, make it more clear?

Something like `descriptor.MessageHeaders.Where(header => !DefaultHeaders.Contains(header));`

Reply:
Or an extension method filtering default headers out, something like `descriptor.MessageHeaders.ExcludeDefaultHeaders()` or `descriptor.MessageHeaders.WithoutDefaultHeaders()`.

Reply:
There may be better ways to implement this. I did not focus more on this than fixing what was broken. 
My main focus for this pull request is differently and I would like to keep that focus. Is that ok for now?